tests/installation/resolve_dependency_issues.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Unified dependency issues resolver
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "y2logsstep";
+use testapi;
+
+sub run {
+    my ($self) = @_;
+    if (check_screen('dependency-issue', 5) && get_var("WORKAROUND_DEPS")) {
```

Comment:
''installation_overview_before' module had the line `assert_screen('installation-settings-overview-loaded', 90);` which allows to immediately skip the module if there are no dependency issues. In the current solution the test will wait 5 and 30 seconds every time because of check_screen with the respective timeouts. 

I guess we should keep the assert_screen line.

Reply:
if i get it the assert_screen('installation-settings-overview-loaded', 90); says everything is fine so exit. without it right now, if the needles are not match the module exit immediately. i can see why we need the assert_screen.

Reply:
I can see potential problem, that page is not yet loaded, so needle doesn't match and then we have dependency issues. In the installation_overview_before we had this check, so this needle only assures that page was loaded and not that everything is fine. Therefore, we need to assert that we are on the correct page first before detecting any issues.

Reply:
30 seconds will be worse, as in case there are no dependency issues we will wait 30 seconds for nothing. In case we have asserted that page was loaded, we can just check if label is there.
Alternative would be multi-tag assert/check screen, than we can bump timeout. But in this particular case it will be tricky as `installation-settings-overview-loaded` will much anyway, even there are dependency issues.

Reply:
@rwx788 can you check the latest submission. i run it just once http://aquarius.suse.cz/tests/694#. i might need to run it a couple of times more if you agree with the approach

Reply:
To be honest, I would just add `assert_screen` call as @OleksandrOrlov has proposed and merge it. With current code it's more changes to the functional part.